### PR TITLE
Update dotnet linters to .NET 8

### PR DESCRIPTION
**Pull request by bdovaz**

### CODE REVIEW

**Review by echoix — APPROVED** [review]

**Comment by echoix** [comment]
At least if we have dotnet 7 in a first release it's already better

**Comment by bdovaz** [comment]
> At least if we have dotnet 7 in a first release it's already better

I already predicted that something might happen and that's why I didn't mix PRs 🤣

https://github.com/oxsecurity/megalinter/pull/2402#issuecomment-1837503011

**Comment by nvuillam** [comment]
Let's quitely wait .net8 to be in alpine packages, I'm so glad we don't need to install it the messy way anymore ^^

**Comment by echoix** [comment]
https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/49831

**Comment by bdovaz** [comment]
It is still the same, I have reported it to see if I call attention and they can prioritize it: https://github.com/dotnet/sdk/issues/37790

**Comment by bdovaz** [comment]
I forgot to mention it, but from what they say, until alpine 3.20 they will not include it: https://github.com/dotnet/sdk/issues/37790#issuecomment-1918068016

**Comment by nvuillam** [comment]
Is there a way to install it from the edge version meanwhile ?

**Comment by bdovaz** [comment]
@nvuillam at least I can't answer you, I have no experience with alpine and its packages.... But in case we can it would be the way to unblock this problem now and in the future because it can happen again with .NET 9...

**Comment by bdovaz** [comment]
Doing a quick search it looks like it is: https://stackoverflow.com/a/52903112

**Comment by nvuillam** [comment]
@bdovaz if you want to try something like `apk add librdkafka --repository=https://dl-cdn.alpinelinux.org/alpine/edge/community`, it will work if you move it from the apk property to the dockerfile property :)

**Comment by bdovaz** [comment]
Done!

File: megalinter/descriptors/vbdotnet.megalinter-descriptor.yml [hunk 1]
```diff
@@ -8,9 +8,9 @@ descriptor_flavors:
 file_extensions:
   - ".vb"
 install:
-  apk:
-    - dotnet7-sdk
   dockerfile:
+    # https://github.com/dotnet/sdk/issues/37790#issuecomment-1918068016
+    - RUN apk add dotnet8-sdk --repository=https://dl-cdn.alpinelinux.org/alpine/edge/community
```
Comment (echoix) [inline megalinter/descriptors/vbdotnet.megalinter-descriptor.yml:13]:
```suggestion
    - RUN apk add --no-cache dotnet8-sdk --repository=https://dl-cdn.alpinelinux.org/alpine/edge/community
```

Comment (bdovaz) [reply megalinter/descriptors/vbdotnet.megalinter-descriptor.yml]: Done!

**Review by echoix — APPROVED** [review]

**Comment by bdovaz** [comment]
@nvuillam  ready!

**Comment by echoix** [comment]
I think it's ok, since the tests pass. How breaking of a change is this to projects using Megalinter (estimated)?

**Review by nvuillam — APPROVED** [review]
Amazing @bdovaz :)

**Pull request merged**